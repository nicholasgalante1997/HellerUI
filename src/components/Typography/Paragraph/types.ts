import { SafeReactFC, WithCustomStyles } from '../../../@types';

export type NativeParagraphProps = {
  color?: string;
  fontSize?: number;
  bold?: boolean;
  italic?: boolean;
  thin?: boolean;
};

export type ParagraphProps = WithCustomStyles<
  SafeReactFC<NativeParagraphProps>
> & { dataTestId?: string };
