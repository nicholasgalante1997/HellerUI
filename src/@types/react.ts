import React from 'react';

export type SafeReactFC<T> = T & {
    children?: React.ReactNode | React.ReactNode[];
};
