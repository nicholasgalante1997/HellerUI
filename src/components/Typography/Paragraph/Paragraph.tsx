import React from 'react';

import {
  Paragraph,
  ParagraphBold,
  ParagraphItalic,
  ParagraphBoldItalic,
  ParagraphThin
} from './Styles';
import { ParagraphProps } from './types';

export default (props: ParagraphProps) => {
  let ContentWrapper = Paragraph;
  const {
    bold = false,
    color,
    customStyles = {},
    fontSize,
    children,
    italic = false,
    thin = false,
    dataTestId = 'h-p-tag'
  } = props;
  if (bold) ContentWrapper = ParagraphBold;
  if (italic) ContentWrapper = ParagraphItalic;
  if (italic && bold) ContentWrapper = ParagraphBoldItalic;
  if (thin) ContentWrapper = ParagraphThin;

  return (
    <ContentWrapper
      data-testid={dataTestId}
      fontSize={fontSize}
      color={color}
      style={customStyles}
    >
      {children}
    </ContentWrapper>
  );
};
