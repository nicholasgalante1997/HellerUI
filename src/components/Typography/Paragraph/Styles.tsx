import styled from 'styled-components';
import { NativeParagraphProps } from './types';

export const Paragraph = styled.p<NativeParagraphProps>`
    color: ${(props: NativeParagraphProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Regular';
    font-style: normal;
    font-size: ${(props) => (props.fontSize ? `${props.fontSize}px` : '1rem')};
    font-weight: 200;
    margin-block-start: 0.25em;
    margin-block-end: 0.25em;
`;

export const ParagraphItalic = styled.p<NativeParagraphProps>`
    color: ${(props: NativeParagraphProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Italic';
    font-style: normal;
    font-size: ${(props) => (props.fontSize ? `${props.fontSize}px` : '1rem')};
    font-weight: 200;
    margin-block-start: 0.25em;
    margin-block-end: 0.25em;
`;

export const ParagraphBold = styled.p<NativeParagraphProps>`
    color: ${(props: NativeParagraphProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Bold';
    font-style: normal;
    font-size: ${(props) => (props.fontSize ? `${props.fontSize}px` : '1rem')};
    font-weight: 800;
    margin-block-start: 0.25em;
    margin-block-end: 0.25em;
`;

export const ParagraphBoldItalic = styled.p<NativeParagraphProps>`
    color: ${(props: NativeParagraphProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins BoldItalic';
    font-style: normal;
    font-size: ${(props) => (props.fontSize ? `${props.fontSize}px` : '1rem')};
    font-weight: 800;
    margin-block-start: 0.25em;
    margin-block-end: 0.25em;
`;

export const ParagraphThin = styled.p<NativeParagraphProps>`
    color: ${(props: NativeParagraphProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Thin';
    font-style: normal;
    font-size: ${(props) => (props.fontSize ? `${props.fontSize}px` : '1rem')};
    font-weight: 500;
    margin-block-start: 0.25em;
    margin-block-end: 0.25em;
`;
