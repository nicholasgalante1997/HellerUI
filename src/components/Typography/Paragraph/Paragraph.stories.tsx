/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Meta, Story } from '@storybook/react';
import Paragraph from './Paragraph';
import { ParagraphProps } from './types';
import '../../../index.css';

export default {
  title: 'Typography/Paragraph',
  component: Paragraph,
  argTypes: {},
  parameters: {
    componentSubtitle: 'a subsection of Typography for regular paragraph text.',
  },
} as Meta;

const Template: Story<ParagraphProps> = (args) => (
  <Paragraph {...args}>The small yellow fox jumped over the moon.</Paragraph>
);

export const Default = Template.bind({});
Default.args = {
  fontSize: 24,
  color: 'violet',
  bold: false,
  italic: false,
};
