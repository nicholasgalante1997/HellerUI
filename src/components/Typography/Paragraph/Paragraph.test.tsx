/* eslint-disable no-restricted-syntax */
/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render } from '@testing-library/react';
import Paragraph from './Paragraph';

describe('Paragraph', () => {
  test('default', () => {
    const { container } = render(<Paragraph>Heller UI</Paragraph>);
    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle('font-family: \'Poppins Regular\';');
    expect(container.firstChild).toHaveStyle('font-weight: 200;');
    expect(container.firstChild).toHaveStyle('font-size: 1rem;');
    expect(container.firstChild).toMatchSnapshot();
  });
  test('dynamic color', () => {
    const { container } = render(<Paragraph color="deeppink">Heller UI</Paragraph>);
    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle('color: deeppink;');
    expect(container.firstChild).toMatchSnapshot();
  });
  test('italic', () => {
    const { container } = render(<Paragraph italic>Heller UI</Paragraph>);
    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle('font-style: italic;');
    expect(container.firstChild).toMatchSnapshot();
  });
  test('bold', () => {
    const { container } = render(<Paragraph bold>Heller UI</Paragraph>);
    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle('font-weight: 800;');
    expect(container.firstChild).toMatchSnapshot();
  });
  test('italic & bold', () => {
    const { container } = render(<Paragraph italic bold>Heller UI</Paragraph>);
    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle('font-style: oblique 45deg;');
    expect(container.firstChild).toHaveStyle('font-weight: 800;');
    expect(container.firstChild).toMatchSnapshot();
  });
});
