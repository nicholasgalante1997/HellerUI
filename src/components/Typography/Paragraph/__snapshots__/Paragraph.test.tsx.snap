// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph bold 1`] = `
<p
  class="sc-dkzDqf bAZhzJ"
>
  Heller UI
</p>
`;

exports[`Paragraph default 1`] = `
<p
  class="sc-bczRLJ BEsZF"
>
  Heller UI
</p>
`;

exports[`Paragraph dynamic color 1`] = `
<p
  class="sc-bczRLJ eKyWXa"
  color="deeppink"
>
  Heller UI
</p>
`;

exports[`Paragraph italic & bold 1`] = `
<p
  class="sc-hKMtZM iAviBA"
>
  Heller UI
</p>
`;

exports[`Paragraph italic 1`] = `
<p
  class="sc-gsnTZi dmboij"
>
  Heller UI
</p>
`;
